File: tests/nodejs/cases/case_031_archive_profile.js
/**
 * Task case 031: archive_profile
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case031ArchiveProfile(input) {
  // Task: archive profile records and prepare transport-ready payload.
  const taskName = 'archive_profile';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'profile',
    stage: 'archive',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is promoted to module-level retained cache for cross-request reuse.
  retainedCache.set('case_031', payload);
  return 'ok';
}

module.exports = {
  case031ArchiveProfile,
};

File: tests/python/cases/case_031_archive_profile.py
"""
Task case 031: archive profile payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_031_archive_profile(input_data):
    # Task: archive profile records and prepare transport-ready payload.
    task_name = 'archive_profile'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'profile',
        'stage': 'archive',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is promoted to module-level RETAINED_CACHE for cross-request reuse.
    RETAINED_CACHE['case_031'] = payload
    return 'ok'
